tests/conftest.py

```diff
@@ -47,6 +51,14 @@ async def app(scope: dict, receive: typing.Callable, send: typing.Callable) -> N
     await send({"type": "http.response.body", "body": b"Hello, world!"})
 
 
+async def slow_app(
```

Comment:
Actually, let's consider updating `app` so it just switches on the requested `path`:

```python
async def app(scope, receive, send):
    path = scope["raw_path"]
    if path.startswith(b"/slow"):
        await slow(scope, receive, send)
    else:
        await hello_world(scope, receive, send)


async def hello_world(scope, receive, send):
    ...


async def slow(scope, receive, send):
    path = scope["raw_path"]
    seconds = int(path.split(b"/")[1].decode())
    await trio.sleep(seconds)
    await hello_world(scope, receive, send)
```

=> Then we can just reuse the existing server fixtures — no need to setup anything new.

Reply:
Good idea. I've done with it